nux/components/dot-tip/index.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import { defer, partial } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, createRef, compose } from '@wordpress/element';
+import { Popover, Button, IconButton } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { withSelect, withDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+export class DotTip extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.popoverRef = createRef();
+	}
+
+	componentDidMount() {
+		if ( this.props.isVisible ) {
+			// Fix the tip not appearing next to the inserter toggle by forcing Popover
+			// to recalculate its size and position on the next frame
+			defer( () => {
+				const popover = this.popoverRef.current;
+				const popoverSize = popover.updatePopoverSize();
+				popover.computePopoverPosition( popoverSize );
+			} );
+		}
+	}
+
+	render() {
+		const { children, isVisible, hasNextTip, onDismiss, onDisable } = this.props;
+
+		if ( ! isVisible ) {
+			return null;
+		}
+
+		return (
+			<Popover
+				ref={ this.popoverRef }
+				className="nux-dot-tip"
+				position="middle right"
+				noArrow
+				focusOnMount
+				role="dialog"
```

Comment:
Should it not be the responsibility of `Popover` to assign this? `aria-modal` ? `aria-label` ?

Reply:
This is a little out of my comfort zone and broaches on the work that @afercia is doing with building a modal component.

Still, my thoughts:

- `Popover` could always set `role="dialog"` because "The dialog role is used to mark up an HTML based application dialog or window that separates content or UI from the rest of the web application or page." ([source](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_dialog_role)) suggests to me that all popovers are dialogs
- `Popover` could set `aria-modal` to `"true"` if there is no `onClose` prop provided
- `Popover` _should_ always have an `aria-label`, but the caller would need to provide it since there isn't a sensible default value